### PR TITLE
qa: Wait for purge to complete 

### PR DESCRIPTION
**Pull request by kotreshhr**
Wait for purge to complete in the test test_volume_info_pending_subvol_deletions

Fixes: https://tracker.ceph.com/issues/62278

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: qa/tasks/cephfs/test_volumes.py [hunk 1]
```diff
@@ -668,14 +668,16 @@ def test_volume_info_pending_subvol_deletions(self):
         subvolname = self._generate_random_subvolume_name()
         # create subvolume
         self._fs_cmd("subvolume", "create", self.volname, subvolname, "--mode=777")
-        # create 3K files of 0.1MB
-        self._do_subvolume_io(subvolname, number_of_files=3000, file_size=0.1)
+        # create 3K zero byte files
+        self._do_subvolume_io(subvolname, number_of_files=3000, file_size=0)
```
Comment (kotreshhr) [inline qa/tasks/cephfs/test_volumes.py:672]: file_size is getting formatted to int before passing to `dd` command. Hence intentionally passing count as 0

**Comment by kotreshhr** [comment]
jenkins test make check arm64

**Comment by kotreshhr** [comment]
jenkins test api

**Comment by kotreshhr** [comment]
jenkins test api

**Review by joscollin — APPROVED** [review]
Hope this should go along with the same backport PRs?

File: qa/tasks/cephfs/test_volumes.py (hunk 1 above)
Comment (vshankar) [reply qa/tasks/cephfs/test_volumes.py:672]: I didn't understand this o_O

**Review by vshankar — APPROVED** [review]
Otherwise LGTM

File: qa/tasks/cephfs/test_volumes.py (hunk 1 above)
Comment (kotreshhr) [reply qa/tasks/cephfs/test_volumes.py:672]: @vshankar The `file_size` is passed to `self.mount_a.write_n_mb`  as `n_mb`. See below that `n_mb` is formatted to int which is a count to `dd` cmd. So essentially this interface can't create a file smaller than 1MB

```
  def _do_subvolume_io(self, subvolume, subvolume_group=None, create_dir=None,
                         number_of_files=DEFAULT_NUMBER_OF_FILES, file_size=DEFAULT_FILE_SIZE):
        ...
        ...
        for i in range(number_of_files):
            filename = "{0}.{1}".format(TestVolumes.TEST_FILE_NAME_PREFIX, i)
            self.mount_a.write_n_mb(os.path.join(io_path, filename), file_size)

```
```
    def write_n_mb(self, filename, n_mb, seek=0, wait=True):
        """
        Write the requested number of megabytes to a file
        """
        assert(self.is_mounted())

        return self.run_shell(["dd", "if=/dev/urandom", "of={0}".format(filename),
                               "bs=1M", "conv=fdatasync",
                               "count={0}".format(int(n_mb)),
                               "seek={0}".format(int(seek))
                               ], wait=wait)

```

Comment (dparmar18) [reply qa/tasks/cephfs/test_volumes.py:672]: shouldn't we then try create files of 1MB then? in order to make sure everything goes as planned even for files with some data in it?

Comment (kotreshhr) [reply qa/tasks/cephfs/test_volumes.py:672]: No, not required for this test case. It simple occupies 3GB of memory.

**Comment by kotreshhr** [comment]
> Hope this should go along with the same backport PRs?

Yes right.

**Review by dparmar18 — APPROVED** [review]

**Pull request merged**